src/interaction/selection.ts

```diff
@@ -13,12 +13,16 @@ let timeout: number = null;
 
 export function start(): void {
     reset();
-    bind(document, "selectstart", recompute, true);
-    bind(document, "selectionchange", recompute, true);
 }
 
-function recompute(): void {
-    let current = document.getSelection();
+export function observe(root: Node): void {
+    bind(root, "selectstart", recompute.bind(this, root), true);
+    bind(root, "selectionchange", recompute.bind(this, root), true);
+}
+
+function recompute(root: Node): void {
+    let doc = root.nodeType === Node.DOCUMENT_NODE ? root as Document : document;
```

Comment:
This is window.document? Would be clearer if written that way (to me0.

Reply:
This could also be iframe.contentDocument too and that's why we are checking if we should bind to a different document, and not the window.document. In other cases (e.g. shadowDom) we bind to "document" which is same as "window.document".